flang/test/Lower/omp-barrier.f90

```diff
@@ -0,0 +1,24 @@
+! This test checks lowering of OpenMP Barrier Directive.
+
+! RUN: bbc -fopenmp -emit-fir %s -o - | \
+! RUN:   FileCheck %s --check-prefix=FIRDialect
+! RUN: bbc -fopenmp -emit-llvm %s -o - | \
+! RUN:   FileCheck %s --check-prefix=LLVMIRDialect
+! RUN: bbc -fopenmp -emit-llvm %s -o - | \
```

Comment:
Typical usage is `bbc` to test lowering Fortran to MLIR, testing the bridge. `tco` to convert MLIR to LLVM IR, testing the optimization passes.  (`bbc` can emit LLVM IR, but it very likely won't drive the optimizer.)